File: practicas/p1/c++/main-dynamic-stack.cc
#include "dynamic-stack.h"
#include <iostream>

int main(int argc,char* argv[])
{
	dynamic_stack<int> stk;

	stk.push(42);
	stk.push(15);
	stk.pop();
	
	for (int i = 0; i < 100; i+=5)
		stk.push(i);
	
	dynamic_stack<int>::iterator iter = stk.begin();
	
	while(iter != stk.end())
	{
		std::cout << *iter << " ";
		++iter;
	}
	std::cout << std::endl;
	
	
	// Opcionalmente, con esta nomenclatura, puede hacerse asi (bucle foreach)
	for (int i : stk)
		std::cout<< i <<" ";
	std::cout << std::endl;
	

	return 0;
}

File: practicas/p1/java/MainDynamicStack.java
import java.util.Iterator;

public class MainDynamicStack
{
	public static void main(String[] args)
	{
		DynamicStack<Integer> stk = new DynamicStack<Integer>();

		stk.push(42);
		stk.push(15);
		stk.pop();
		for (int i=0; i<100; i+=5)
			stk.push(i);

		Iterator<Integer> iter = stk.iterator();
		while(iter.hasNext())
			System.out.print(iter.next()+" ");
		System.out.println();

		// Opcionalmente, con esta nomenclatura, puede hacerse asi
		for (Integer i : stk)
			System.out.print(i+" ");
		System.out.println();
	}
}
